### PR TITLE
Enhance OS version parsing to handle SUSE SP notation

### PR DESCRIPTION
**Pull request by sidis-zafran**
## Description
This PR enhances the `versionGrok` regex pattern within makeOSInfo to support SUSE Linux Enterprise Server (**SLES**) versions that use `SP` notation for Service Packs, such as `15-SP5`.

The updated regex now recognizes `SP` as a valid separator for the minor version, allowing accurate parsing of **SLES** versions alongside standard major.minor.patch formats (e.g., `22.04`). This change resolves issues where Minor was incorrectly set to `0` for `SP`-based versions and ensures compatibility with both SUSE-specific and conventional version formats.

## Changes
Updated `versionGrok` regex: Added optional SP notation support as a prefix for the minor version.
Compatibility maintained: Supports both `.` and `-` delimiters to handle multiple versioning schemes.
## Testing
Test cases added: Included tests for both `SLES 15-SP5` and `SLED 15-SP5` to validate correct assignment of major, minor, and patch values.

### CODE REVIEW

**Review by MaorRayzinZ — APPROVED** [review]
looks good (not that anybody can read regex)

**Comment by andrewkroh** [comment]
> not that anybody can read regex

That also means the maintainability of the regex will suffer. I would like to see another revision where the code is modified to have a separate regex for suse. I think is will be more readable and better to maintain. Then either we try each regex to see which one matches OR we use the ID_LIKE="suse" to choose the right regex to apply.

**Comment by sidis-zafran** [comment]
> > not that anybody can read regex
> 
> That also means the maintainability of the regex will suffer. I would like to see another revision where the code is modified to have a separate regex for suse. I think is will be more readable and better to maintain. Then either we try each regex to see which one matches OR we use the ID_LIKE="suse" to choose the right regex to apply.

Totally agreed. Updated with slight changes.

**Comment by sidis-zafran** [comment]
@andrewkroh 
Added release-note. Please rerun checks so we may merge this one 🙏

File: .changelog/248.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
+Support SLES/SLED version extraction
```
Comment (andrewkroh) [inline .changelog/248.txt:2]:
```suggestion
linux: Support SLES/SLED OS version extraction.
```

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**